File: app/src/main/java/java/android/notes/core/Control.java
package java.android.notes.core;

import java.android.notes.Time;

public class Control {
    private enum Mode{ addNote,openNote }
    private Mode mode = Mode.addNote;

    public Notes notes = new Notes();
    public Note note = new Note(Time.getDateYMD());

    public void createNote() {
        note = new Note(Time.getDateYMD());
        mode = Mode.addNote;
    }

    public void openNoteOutList(int index) {
        note = notes.getNote(index);
        mode = Mode.openNote;
    }

    public void sortedNotes(){
        notes.sorted();
    }

    public void removeNoteOutList(int index) {
        notes.removeNote(notes.getNote(index));
    }

    public boolean save(String headline,String description,String body){
        if(headline.equals("")){ return false; }

        note.setHeadline(headline);
        note.setDescription(description);
        note.setBody(body);

        if(mode == Mode.addNote){
            notes.addNote(note);
        }

        return true;
    }

}

File: app/src/main/java/java/android/notes/core/Control.kt
package java.android.notes.core

import java.android.notes.Time

class Control {
    private enum class Mode {
        addNote, openNote
    }

    private var mode = Mode.addNote

    var notes = Notes()

    var note: Note = Note(Time.dateYMD)
    fun createNote() {
        note = Note(Time.dateYMD)
        mode = Mode.addNote
    }

    fun openNoteOutList(index: Int) {
        note = notes.getNote(index)
        mode = Mode.openNote
    }

    fun sortedNotes() {
        notes.sorted()
    }

    fun removeNoteOutList(index: Int) {
        notes.removeNote(notes.getNote(index))
    }

    fun save(headline: String, description: String, body: String): Boolean {
        if (headline == "") {
            return false
        }

        note.setterHeadLine(headline)
        note.setterDescription(description)
        note.setterBody(body)
        if (mode == Mode.addNote) {
            notes.addNote(note)
        }
        return true
    }
}